pandas/core/arrays/categorical.py

```diff
@@ -2439,11 +2439,15 @@ def _get_codes_for_values(values, categories):
     """
     utility routine to turn values into codes given the specified categories
     """
-
     from pandas.core.algorithms import _get_data_algo, _hashtables
     if not is_dtype_equal(values.dtype, categories.dtype):
         values = ensure_object(values)
```

Comment:
I can do that if you want. One thing to note is that I believe `if not is_dtype_equal` case is probably the "common case", since it doesn't occur with many `dtype`s like integers or strings.

I don't *think* there's any speed advantage to being in the `if` branch vs. the `else` branch, but depending on convention you may prefer the common case to be the first block and the rare case to be the `else` block. Up to you.

Reply:
its not about speed or anything else, its about consistency in our code of how we do these kinds of tests.